src/lib.rs

```diff
@@ -365,189 +367,81 @@ where
-                value: arg,
-            }),
-        }
-    }
-
-    /// Retrieves the rest of the positional arguments, after the
-    /// options have been parsed.
-    ///
-    /// This method returns the list of arguments after the parsed
-    /// options.
-    ///
-    /// # Panics
-    ///
-    /// This method panics if the option parsing is not yet complete;
-    /// that is, it panics if [`next`](#method.next) has not yet
-    /// returned `None` at least once.
+    /// # use getargs::{Opt, Options};
+    /// #
+    /// let args = ["--flag", "one", "two"];
+    /// let mut opts = Options::new(args.into_iter());
     ///
-    /// # Examples
+    /// assert_eq!(opts.next(), Ok(Some(Opt::Long("flag"))));
+    /// assert_eq!(opts.next(), Ok(None));
     ///
-    /// Basic usage:
+    /// let mut args = opts.args();
     ///
+    /// assert_eq!(args.next(), Some("one"));
+    /// assert_eq!(args.next(), Some("two"));
+    /// assert_eq!(args.next(), None);
     /// ```
-    /// use getargs::{Opt, Options};
-    /// let args = ["-a", "foo", "bar"];
-    /// let opts = Options::new(&args);
-    /// assert_eq!(opts.next(), Ok(Some(Opt::Short('a'))));
-    /// assert_eq!(opts.next(), Ok(None));
-    /// assert_eq!(opts.args(), &["foo", "bar"]);
-    /// ```
-    pub fn args(&self) -> &'a [S] {
-        let inner = self.inner.borrow();
-        match inner.state {
-            State::Start => &self.args[inner.position..],
-            _ => panic!("called args() while option parsing hasn't finished"),
-        }
+    pub fn args<'opts>(&'opts mut self) -> ArgIterator<'opts, 'str, I> {
+        ArgIterator::new(self)
     }
-}
-
-/// A short or long option.
-#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
-pub enum Opt<'a> {
-    /// A short option, as in `-a`.
-    Short(char),
-    /// A long option, as in `--attack`.
-    Long(&'a str),
-}
 
-impl Display for Opt<'_> {
-    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        match self {
-            Opt::Short(c) => write!(f, "-{}", c),
-            Opt::Long(s) => write!(f, "--{}", s),
+    /// "Restarts" options parsing if the iterator has been exhausted.
+    /// This only results in any noticeable effect if the iterator is a
+    /// repeating iterator; otherwise, nothing happens.
+    ///
+    /// A "repeating iterator" is one that starts to produce elements
+    /// again even after a call to [`Iterator::next`] returns `None`.
+    pub fn restart(&'_ mut self) {
+        self.state = match self.state {
+            State::End => State::Start,
+            _ => panic!("called Options::restart() during an iteration"),
         }
     }
 }
 
-impl From<char> for Opt<'static> {
-    fn from(ch: char) -> Opt<'static> {
-        Opt::Short(ch)
-    }
-}
-
-impl<'a> From<&'a str> for Opt<'a> {
-    fn from(s: &'a str) -> Opt<'a> {
-        Opt::Long(s)
-    }
-}
-
-// Error handling
+impl<'str, I: Iterator<Item = &'str str>> IntoIterator for Options<'str, I> {
```

Comment:
I don't really want to add an IntoIterator impl -- it seems like it should do the same thing as `next()`, but it doesn't. Maybe call it `into_args()` instead.